モデルデータの利用条件表記ガイドラインに合わせて利用条件表記を更新

ke456-png opened the pull request:
モデルデータの利用条件表記ガイドラインが更新されたため、exampleでの利用条件表記を変更しました。
https://developer.vroid.com/guidelines/conditions_of_use.html

 一部の変更行にフォーマッタによる変更を含みます。

## スクリーンショット

### 旧 ↔ 新
<img height="600" alt="image" src="https://github.com/user-attachments/assets/2214fffa-6552-4f0d-bd27-010a03fb86b0" /><img height="600" alt="image" src="https://github.com/user-attachments/assets/6126c395-7c92-4d39-a7fb-236fee513253" />





## 関連ドキュメント

モデルデータの利用条件表記ガイドライン
https://developer.vroid.com/guidelines/conditions_of_use.html

ガイドライン内の表示例
<img width="1376" height="800" alt="image" src="https://github.com/user-attachments/assets/1ef7b8e3-7e45-4e83-920b-071f169de142" />

ke456-png committed:
絶対パスでのインポートとエイリアスの設定

ke456-png committed:
モデルデータの利用条件表記ガイドラインに合わせ、利用条件表記を更新する

モデルデータの利用条件表記ガイドラインが更新されたため、exampleでの利用条件表記を更新します。
https://developer.vroid.com/guidelines/conditions_of_use.html

- これまでUI向けにexample側で定義していたModelData型を廃止し、CharacterModelSerializerに型を統一しました。
- VRoid Hubで設定できるVRM0用のライセンスの表示を更新しました。
- VRM1用のライセンス表示を更新しました。

一部の変更行にフォーマッタによる変更を含みます。

ke456-png committed:
fix color token

ke456-png committed:
fix imports

0b5vr commented on components/ModelLicense/HubVrm0License.tsx:
```diff
@@ -0,0 +1,56 @@
+import { CharacterModelLicenseSerializer } from '@/types/Response';
```
ここも `type` を付与できそうです

ke456-png replied:
ありがとうございます、https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/cfc652531b2f3afd50444dc75802be4204e3132f で本PRに含まれる別ファイルも合わせて修正しました。

ke456-png committed:
fix type-only imports

0b5vr reviewed (approved):
Looks good

nibankougen commented on components/ModelLicense/licenseUtils.ts:
```diff
@@ -0,0 +1,175 @@
+    case 'characterization_allowed_user':
+      return status === 'everyone' ? 'allow' : 'default';
+    case 'personal_commercial_use':
+      return status === 'disallow' || status === 'default' ? 'default' : 'allow';
```
ここだけ`'default'`がデフォルト扱いになっていない点が気になりました。
他の項目ように必要な要素のみ`'allow'`にするのであればのようになるかと思います。

 
```suggestion
      return status === 'profit' || status === 'nonprofit' ? 'allow' : 'default'
```

ke456-png replied:
https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/2e31e828648c23ec6be4e06a4816b528bed9b64b にて修正しました

nibankougen commented on components/ModelLicense/licenseUtils.ts:
```diff
@@ -0,0 +1,175 @@
+  //「 個人の商用利用」は法人利用とプロパティが共用のため、commercialUsageの値を参照し判定する
+  if (property === 'personalCommercialUse') {
+    const status = vrm1Meta.commercialUsage;
+    if (typeof status === 'undefined') return '未設定';
```
ここは文字列でなく、undefinedだった場合の想定です。

```suggestion
    if (typeof status === undefined) return '未設定';
```

three-vrmのVRM1Metaの定義上もundefinedを許容する型定義になっています

https://github.com/pixiv/three-vrm/blob/9a1814ade92e2617cb4b96e6210e15e45e2ff935/packages/three-vrm-core/src/meta/VRM1Meta.ts#L74C3-L74C18

ke456-png replied:
typeof演算子を使っているので一応意図通りにはなっているはずです。
ただ、undefined | false | trueのプロパティがあったのでこのように実装しましたが、`status == null`の方が良かったかもしれないです。

nullを区別したい項目はないですよね？

nibankougen replied:
失礼しました。確かにtypeofがついていました。

nullを区別したい項目はないですね。

ke456-png replied:
https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/7319d0476ce631ea52d6e8bc2c1ba69efacbb7d0 にて`status == null`へ変更しました。

nibankougen commented on components/ModelLicense/licenseUtils.ts:
```diff
@@ -0,0 +1,175 @@
+  // 「改変物の再配布」は改変とプロパティが共用のため、modificationの値を参照し判定する
+  if (property === 'modificationRedistribution') {
+    const status = vrm1Meta.modification;
+    if (typeof status === 'undefined') return '未設定';
```
ここも同様です

```suggestion
    if (typeof status === undefined) return '未設定';
```

nibankougen commented on components/ModelLicense/licenseUtils.ts:
```diff
@@ -0,0 +1,175 @@
+  }
+
+  const status = vrm1Meta[property];
+  if (typeof status === 'undefined') return '未設定';
```
ここも同様です

```suggestion
  if (typeof status === undefined) return '未設定';
```

nibankougen commented on components/ModelLicense/licenseUtils.ts:
```diff
@@ -0,0 +1,175 @@
+    case 'modification':
+      return status === 'allowModification' || status === 'allowModificationRedistribution' ? 'OK' : 'NG';
+    case 'creditNotation':
+      return status === 'necessary' ? '必要' : '不要';
```
ここは`'required'`です
参考: https://github.com/pixiv/three-vrm/blob/9a1814ade92e2617cb4b96e6210e15e45e2ff935/packages/three-vrm-core/src/meta/VRM1Meta.ts#L89

```suggestion
      return status === 'required' ? '必要' : '不要';
```

ke456-png replied:
見落としていました！修正します。

HubVRM0では"necessary" / "unnecessary" 、VRM1では"required"/ "unnecessary"ですね！

nibankougen replied:
はい！その通りです。

ke456-png replied:
https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/b010d48e05c639cf63ebc870470957bdd7a606f2 にて修正しました

ke456-png committed:
licenseでのdefaultの扱いを修正

personal_commercial_useの分岐とデフォルト扱いを他の箇所へ揃えました。
VRM0においてもライセンスの定義外の場合、VRM1同様にdefaultを返すように変更しました。

ke456-png committed:
VRM1ライセンスのcreditNotationを修正

requiredとすべきところがnecessaryとなっていました。

ke456-png committed:
未設定の分岐を修正

ライセンスガイドラインにおいてnullとundefinedを区別する必要はないため修正しました。

undefined | true | false のケースは存在するので注意が必要

nibankougen reviewed (approved):
LGTM
